openapi/openapi/src/main/scala/endpoints4s/openapi/Urls.scala

```diff
@@ -48,13 +49,49 @@ trait Urls extends algebra.Urls {
       )
     )
 
-  case class DocumentedQueryStringParam(schema: Schema, isRequired: Boolean)
-  type QueryStringParam[A] = DocumentedQueryStringParam
+  override def optQsWithDefault[A](name: String, default: A, docs: Documentation = None)(implicit
+      value: QueryStringParam[Option[A]]
+  ): QueryString[A] =
+    DocumentedQueryString(
+      List(
+        DocumentedParameter(
+          name,
+          required = false,
+          docs,
+          schema = {
+            def withDefault(defaultValue: Option[ujson.Value]) = value.schema match {
+              case s: Schema.Object    => s.withDefault(defaultValue)
+              case s: Schema.Array     => s.withDefault(defaultValue)
+              case s: Schema.Enum      => s.withDefault(defaultValue)
+              case s: Schema.Primitive => s.withDefault(defaultValue)
+              case s: Schema.OneOf     => s.withDefault(defaultValue)
+              case s: Schema.AllOf     => s.withDefault(defaultValue)
+              case s: Schema.Reference => s
+            }
+
+            withDefault(Some(value.encoder.encode(Some(default))))
+          }
+        )
+      )
+    )
+
+  case class DocumentedQueryStringParam[A](
+      schema: Schema,
+      isRequired: Boolean,
+      encoder: Encoder[A, ujson.Value]
```

Comment:
I don't think this field should be an `encoder` - it should be fine to just have `default: ujson.Value`. Provided you have a `tpe: JsonSchema[A]` in scope when you create your `DocumentedQueryStringParam`, you can use your `defaultValue: A` and do

```scala
   val default: ujson.Value = Some(tpe.ujsonSchema.encoder.encode(defaultValue))
```

Reply:
There is no `JsonSchema[A]` in scope when `DocumentedQueryStringParam` is created, and that is exactly why I added `encoder` here, to be able to encode `default` value when setting it in schema. I see `encoder` as just what is minimally needed here (`JsonSchema[A]` also provides other features, not needed here at the moment). I'm open to include `JsonSchema[A]` if there is a simpler way than this, but I'm not sure I see it at the moment.

Reply:
Oh right, I see what you mean.